src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/JSException.cs

```diff
@@ -47,9 +47,10 @@ internal JSException(string msg, JSObject? jsException) : base(msg)
                 }
 
 #if FEATURE_WASM_THREADS
-                var currentTID = JSSynchronizationContext.CurrentJSSynchronizationContext?.TargetTID;
-                if (jsException.OwnerTID != currentTID)
+                var ctx = JSProxyContext.CurrentInstance;
+                if (ctx == null || jsException.ProxyContext != ctx)
                 {
+                    // if we are on another thread, it would be too expensive and risky to obtain lazy stack trace.
```

Comment:
should we explicitly put in some text that indicates the stack trace was not captured due to the exception crossing threads, so the user can troubleshoot this? otherwise it's just Mysteriously Missing Stack